vuu-ui/packages/vuu-layout/src/layout-persistence/LocalLayoutPersistenceManager.ts

```diff
@@ -8,72 +8,131 @@ const metadataSaveLocation = "layouts/metadata";
-        return layout[0].json;
-      }
-    }
+  loadLayout(id: string): Promise<LayoutJSON> {
+    return new Promise((resolve, reject) => {
+      this.validateId(id, false)
+        .then(() => this.loadLayouts())
+        .then(existingLayouts => {
+          const layouts = existingLayouts.filter(layout => layout.id === id);
+          resolve(layouts[0].json);
+        })
+        .catch(e => reject(e));
+    });
   }
 
-  loadMetadata(): LayoutMetadata[] {
-    return getLocalEntity<LayoutMetadata[]>(metadataSaveLocation) || [];
+  loadMetadata(): Promise<LayoutMetadata[]> {
+    return new Promise((resolve) => {
+      const metadata = getLocalEntity<LayoutMetadata[]>(metadataSaveLocation);
+      resolve(metadata || []);
+    })
   }
 
-  private loadLayouts(): Layout[] {
-    return getLocalEntity<Layout[]>(layoutsSaveLocation) || [];
+  private loadLayouts(): Promise<Layout[]> {
+    return new Promise((resolve) => {
+      const layouts = getLocalEntity<Layout[]>(layoutsSaveLocation);
+      resolve(layouts || []);
+    });
   }
 
-  private appendAndPersist(newId: string,
-                           newMetadata: Omit<LayoutMetadata, "id">,
-                           newLayout: LayoutJSON,
-                           existingLayouts: Layout[],
-                           existingMetadata: LayoutMetadata[]) {
-    existingLayouts.push({id: newId, json: newLayout});
-    existingMetadata.push({id: newId, ...newMetadata});
+  private appendAndPersist(
+    newId: string,
+    newMetadata: Omit<LayoutMetadata, "id">,
+    newLayout: LayoutJSON,
+    existingLayouts: Layout[],
+    existingMetadata: LayoutMetadata[]
+  ) {
+    existingLayouts.push({ id: newId, json: newLayout });
+    existingMetadata.push({ id: newId, ...newMetadata });
 
     this.saveLayoutsWithMetadata(existingLayouts, existingMetadata);
   }
 
-  private saveLayoutsWithMetadata(layouts: Layout[], metadata: LayoutMetadata[]): void {
+  private saveLayoutsWithMetadata(
+    layouts: Layout[],
+    metadata: LayoutMetadata[]
+  ): void {
     saveLocalEntity<Layout[]>(layoutsSaveLocation, layouts);
     saveLocalEntity<LayoutMetadata[]>(metadataSaveLocation, metadata);
   }
+
+  private async validateIds(id: string): Promise<void> {
+    return Promise
+      .all([
+        this.validateId(id, true).catch(error => error.message),
+        this.validateId(id, false).catch(error => error.message)
+      ])
+      .then((errorMessages: string[]) => {
+        const combinedMessage = errorMessages.filter(Boolean).join("; ");
+        if (combinedMessage) {
+          throw new Error(combinedMessage);
+        }
+      });
+  }
+
+  private validateId(id: string, metadata: boolean): Promise<void> {
+    return new Promise((resolve, reject) => {
+      const loadFunc = metadata ? this.loadMetadata : this.loadLayouts;
+      loadFunc().then(result => {
+        const count = result.filter(x => x.id === id).length;
```

Comment:
I'm getting a typescript error on `result.filter(...)`:

![image](https://github.com/ScottLogic/finos-vuu/assets/99646608/a002a428-6632-49bb-a990-f01fa2f919df)

Reply:
Try making Layout and LayoutMetadata follow a common interface.

Reply:
Changed layout types in da474c05; used the new `WithId` interface in 9572cb78.
I've also installed ESLint now, so hopefully will pick this sort of thing up in future.

Reply:
I don't love `WithId` but I also don't have any helpful suggestions 😄 I see how it solves the problem.